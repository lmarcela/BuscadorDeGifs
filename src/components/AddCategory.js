import React, { useState } from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
    const [inputValue, setInputValue] = useState('');
    
    const handleInputChange = (e) =>{   
        setInputValue(e.target.value)
    }
    const handleSubmit = (e) =>{   
        e.preventDefault(); 
        if(inputValue.trim().length>2){
            setCategories( c => [inputValue,...c]);
            setInputValue('');
        }             
    }
    return (
          <form className="w3-container" onSubmit={handleSubmit}>
            <input className="w3-input w3-border" type="text" value={inputValue} onChange={handleInputChange} id="in0"/>  
          </form>
    )
}
    
AddCategory.propTypes={
    setCategories: PropTypes.func.isRequired
}
