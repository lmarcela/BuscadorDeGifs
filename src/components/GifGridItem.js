import React from 'react'
import PropTypes from 'prop-types'
export const GifGridItem = ({title, url}) => {
    return (
        <>        
            <div className="w3-third">
                <div className="w3-card w3-center animate__animated animate__slideInUp animate__faster">
                    <img src={url} alt={title} style={{height:200, maxWidth:"-webkit-fill-available"}} />
                    <p>{title?.trim() || "Sin titulo"}</p>
                </div>
            </div>
        </>
    )
}
GifGridItem.propTypes={
    title : PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}
