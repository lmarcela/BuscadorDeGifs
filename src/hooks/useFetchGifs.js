import { useState, useEffect } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = (category) => {
    const [state, setstate] = useState({
        data: [],
        loading: true
    });

    useEffect(() => {
        getGifs(category).then(imagenes=>{
            setTimeout(() => {
                setstate({
                    data: imagenes,
                    loading: false
                })
            }, 3000);
        })
    }, [category])

    return state; // {data: [], loading: true};
}
