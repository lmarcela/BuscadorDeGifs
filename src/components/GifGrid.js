import React from 'react'
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({category}) => {
    const {data:imagenes, loading} = useFetchGifs(category);
    
    return (
        <>
            <h3 className="animate__animated animate__fadeIn">{category}</h3>
            {loading && <p className="animate__animated animate__flash">Loading</p>}
            
            {<div className="w3-row-padding">
                {
                    imagenes.map((img) => (
                        <GifGridItem key={img.id} {...img}/>
                    )) 
                }
            </div>}
            
            {!loading && imagenes.length===0 && <p>Sin resultados</p>}
        </>
    )
}
