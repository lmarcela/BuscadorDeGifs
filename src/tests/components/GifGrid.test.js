import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import {GifGrid} from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas en <GifGrid />',()=>{
    const category = "puppy";
      

    test('debe mostrar <GifGrid /> correctamente', () => {     
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true            
        })
        
        const wrapper=shallow(<GifGrid category={category} />); 
        expect(wrapper).toMatchSnapshot();
    })

    test('debe de mostrar items cuando se cargan imÃ¡genes usando useFetchGifs', () => { 
        const gifs = [{
            id: "ABC",
            title: "Titulo",
            url: "https://imagenes/imagen.jpg"
        },{
            id: "DEF",
            title: "Titulo2",
            url: "https://imagenes/imagen2.jpg"
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false            
        })

        const wrapper=shallow(<GifGrid category={category} />); 

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
        
    })
});


