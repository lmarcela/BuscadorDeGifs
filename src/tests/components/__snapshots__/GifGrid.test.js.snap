// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGrid /> debe de mostrar items cuando se cargan im√°genes usando useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    puppy
  </h3>
  <div
    className="w3-row-padding"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Titulo"
      url="https://imagenes/imagen.jpg"
    />
    <GifGridItem
      id="DEF"
      key="DEF"
      title="Titulo2"
      url="https://imagenes/imagen2.jpg"
    />
  </div>
</Fragment>
`;

exports[`Pruebas en <GifGrid /> debe mostrar <GifGrid /> correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    puppy
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="w3-row-padding"
  />
</Fragment>
`;
